packages/resource-deployment/scripts/restart-vm-pools.sh

```diff
@@ -28,18 +28,19 @@ restartBatchPools() {
     local batchAccountName
     getBatchAccountName batchAccountName
 
-    echo "Querying for vm scalesets resource groups of batch account $batchAccountName"
-    local query="[?tags.BatchAccountName=='$batchAccountName']"
-    vmssResourceGroupNames=$(az vmss list --subscription "$subscription" --query "$query.[resourceGroup]" -o tsv)
+    echo "Querying pools for $batchAccountName"
+    batchPoolIds=$(az batch pool list --account-name "$batchAccountName" --subscription "$subscription" --query "[*].id" -o tsv)
 
-    for vmssResourceGroupName in $vmssResourceGroupNames; do
-        local name
+    for poolId in $batchPoolIds; do
+        local poolNodeIds
 
-        echo "Querying for vm scaleset name under resource group $vmssResourceGroupName"
-        name=$(az vmss list --subscription "$subscription" --resource-group "$vmssResourceGroupName" --query "$query.[name]" -o tsv)
+        echo "Querying node list for poolId $poolId"
+        poolNodeIds=$(az batch node list --account-name "$batchAccountName" --subscription "$subscription" --pool-id "$poolId" --query "[*].id" -o tsv)
 
-        echo "Restarting vm scaleset $name under resource group $vmssResourceGroupName"
-        az vmss restart --name "$name" --resource-group "$vmssResourceGroupName" --subscription "$subscription" 1>/dev/null
+        for nodeId in $poolNodeIds; do
+            echo "Restarting node with nodeId $nodeId under poolId $poolId"
+            az batch node reboot --node-id $nodeId --pool-id $poolId --account-name $batchAccountName --subscription $subscription --node-reboot-option requeue 1>/dev/null
```

Comment:
should you do taskcompletion instead of requeue? what if the executing task is a job manager task? killing that will kill all the child tasks rite?